os/fs/smartfs/smartfs_smart.c

```diff
@@ -298,12 +298,51 @@ static int smartfs_open(FAR struct file *filep, const char *relpath, int oflags,
 
 			/* At this point, either an available entry was found or a new one has been created */
 			newentry.firstsector = data_sector;
+#ifdef CONFIG_SMARTFS_USE_SECTOR_BUFFER
+			/* If CRC is enabled, hold the entry for writing later */
+			if (parentdirsector != newentry.dsector) {
+				/* Mark flags to indicate that the entry will be written to anew sector in the parent directory.
+				   This sector will then be chained in the end.
+				 */
+				sf->bflags |= SMARTFS_BFLAG_NEW_DIRSECTOR;
+				sf->last_psector = parentdirsector;
+			}
 
+			sf->entry.firstsector = newentry.firstsector;
+			sf->entry.dsector = newentry.dsector;
+			sf->entry.doffset = newentry.doffset;
+#ifdef CONFIG_SMARTFS_ALIGNED_ACCESS
```

Comment:
utc number will be increased when it written by writeentry..
Or you can this in find_availableentry, I think.
name also can be set there.
Then I estimate that we can use sf->entry...

Reply:
Updated